server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -136,9 +139,14 @@ public void close() {
         };
     }
 
-    private Mono<MailWithEnqueueId> loadMail(Delivery response) {
+    private Mono<MailWithEnqueueId> loadMail(AcknowledgableDelivery response) {
         return toMailReference(response)
-            .flatMap(mailLoader::load);
+            .flatMap(reference -> mailLoader.load(reference)
+                .onErrorResume(e -> {
+                    LOGGER.error("Fail to load mail {} with enqueueId {}", reference.getName(), reference.getEnqueueId(), e);
+                    response.nack(REQUEUE);
```

Comment:
if there a non-synchronous version of `nack`?

Reply:
No

Reply:
MQ drivers are such a joke

Reply:
I would not generalize that much but I tend to agree.

Reply:
You missed the coffee break, that's why